ambry-quota/src/main/java/com/github/ambry/quota/HelixStorageUsageRefresher.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.quota;
+
+import com.github.ambry.config.StorageQuotaConfig;
+import com.github.ambry.server.StatsSnapshot;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Random;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.helix.AccessOption;
+import org.apache.helix.store.HelixPropertyListener;
+import org.apache.zookeeper.data.Stat;
+import org.apache.helix.store.HelixPropertyStore;
+import org.apache.helix.zookeeper.datamodel.ZNRecord;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * A Helix implementation of {@link StorageUsageRefresher}. A Helix aggregation task, created in ambry-server, would persist
+ * current storage usage in ZooKeeper periodically. This implementation would fetch the storage usage from zookeeper using
+ * {@link HelixPropertyStore} and subscribe to the change. This implementation also periodically refetch the storage usage
+ * from ZooKeeper to keep it up-to-date.
+ */
+public class HelixStorageUsageRefresher implements StorageUsageRefresher {
+  // These two constant fields are shared by other classes. TODO: reuse these two constant fields.
+  public static final String AGGREGATED_CONTAINER_STORAGE_USAGE_PATH = "Aggregated_AccountReport";
+  public static final String VALID_SIZE_FILED_NAME = "valid_data_size";
```

Comment:
This is not configurable. Whatever goes to config would be considered a configurable value. I am thinking about abstract some of the logic in this class out with aggregation task since they share the same znode path and the same format.